_posts/2021-04-22-release-1.13.0.md

```diff
@@ -0,0 +1,450 @@
+
+```sql
+SELECT window_time, ...
+  FROM (
+    SELECT *, ROW_NUMBER() OVER (PARTITION BY window_start, window_end ORDER BY total_price DESC) 
+      as rank 
+    FROM t
+  ) WHERE rank <= 100;
+```
+
+### Improved interoperability between DataStream and Table API/SQL 
+
+The Table API becomes equally important to the DataStream API which is why a good interoperability
+between the two APIs is crucial. (FLIP-136): The core Row class has received a major update with new
+toString/hash/equals methods. Fields can not only be accessed by position but also by name with
+support of sparse representations. The new methods `StreamTableEnvironment.toDataStream/fromDataStream`
+can model the DataStream API as a regular table source or sink with a smooth type system and event-time
+integration.
+
+```java
+Table table=tableEnv.fromDataStream(
+	dataStream,Schema.newBuilder()
+	.columnByMetadata("rowtime","TIMESTAMP(3)")
+	.watermark("rowtime","SOURCE_WATERMARK()")
+	.build());
+
+DataStream<Row> dataStream=tableEnv.toDataStream(table)
+	.keyBy(r->r.getField("user"))
+	.window(...)
+```
+
+### Improved SQL Client 
+
+During the past releases, the SQL Client's feature set could not keep up with the evolution of the
+TableEnvironment in the Table API. In this release, the SQL Client has been updated for feature
+parity with the programmatic APIs.
+
+*Easier Configuration and Code Sharing*
+
+The support of YAML files to configure the SQL Client will be discontinued. Instead, the client
+accepts one or more initialization scripts to configure an empty session. Additionally, a user can
+declare a main SQL script for executing multiple statements in one file.
+
+```
+./sql-client.sh -i init1.sql init2.sql -f sqljob.sql
+```
+
+*More Commands and Support for Statement Sets*
+
+For multi-statement execution, commands such as ADD JAR, STATEMENT SET, and improved SET/RESET
+commands with new config options have been added. The following example shows how a full SQL script
+could be used as a command line tool to submit Flink jobs in a unified way for both batch and
+streaming use cases:
+
+```sql
+-- set up a catalog
+CREATE
+CATALOG hive_catalog WITH (
+'type' = 'hive'
+);
+USE
+CATALOG hive_catalog;
+
+-- or use temporary objects
+CREATE
+TEMPORARY TABLE users (
+user_id BIGINT,
+user_name STRING,
+user_level STRING,
+region STRING,
+PRIMARY KEY (user_id) NOT ENFORCED
+) WITH (
+'connector' = 'upsert-kafka',
+'topic' = 'users',
+'properties.bootstrap.servers' = '...',
+'key.format' = 'csv',
+'value.format' = 'avro'
+);
+
+-- set the execution mode for jobs
+SET
+execution.runtime-mode=streaming;
+
+-- set the sync/async mode for INSERT INTOs
+SET
+table.dml-sync=true;
+
+-- set the job's parallelism
+SET
+parallism.default=100;
+
+-- restore state from the specific savepoint path
+SET
+execution.savepoint.path=/tmp/flink-savepoints/savepoint-bb0dab;
+
+INSERT INTO user_stats
+SELECT user_id, COUNT(*)
+FROM users
+GROUP BY user_id;
```

Comment:
What about using STATEMENT SET in the example? I prepared a example as following:

```sql
-- set up a catalog
CREATE CATALOG hive_catalog WITH ('type' = 'hive');
USE CATALOG hive_catalog;

-- or use temporary objects
CREATE TEMPORARY TABLE clicks (
  user_id BIGINT,
  page_id BIGINT,
  viewtime TIMESTAMP
) WITH (
  'connector' = 'kafka',
  'topic' = 'clicks',
  'properties.bootstrap.servers' = '...',
  'format' = 'avro'
);

-- set the execution mode for jobs
SET execution.runtime-mode=streaming;

-- set the sync/async mode for INSERT INTOs
SET table.dml-sync=false;

-- set the job's parallelism
SET parallism.default=10;

-- set the job name
SET pipeline.name = my_flink_job;

-- restore state from the specific savepoint path
SET execution.savepoint.path=/tmp/flink-savepoints/savepoint-bb0dab;

BEGIN STATEMENT SET;

INSERT INTO pageview_pv_sink
SELECT page_id, count(1)
FROM clicks
GROUP BY page_id;

INSERT INTO pageview_uv_sink
SELECT page_id, count(distinct user_id)
FROM clicks
GROUP BY page_id;

END;
```